src/app/shell/main/main.component.ts

```diff
@@ -34,31 +41,49 @@ export class MainComponent implements OnInit {
   destroy$: Subject<boolean> = new Subject<boolean>();
   @ViewChild('WorkshopsWrap') workshopsWrap: ElementRef;
   public parent: boolean;
+
   getEmptyCards = Util.getEmptyCards;
   widthOfWorkshopCard = Constants.WIDTH_OF_WORKSHOP_CARD;
-  constructor(
-    private store: Store,
-    private actions$: Actions,
-  ) { }
 
+  constructor(private store: Store) { }
+  
+  getTopWorkshops(role: string): void {
+    if(role === Role.parent) {
+      combineLatest([this.city$, this.favoriteWorkshops$])
+      .pipe(
+        filter((result) => {
+          if(!!result[0] && !!result[1]?.length) {
+            return true;
+          }
+          else if(result[1] === null) {
+            return true;
+          }
+        }),
+        takeUntil(this.destroy$))
+      .subscribe(()=> this.store.dispatch(new GetTopWorkshops(Constants.ITEMS_PER_PAGE)));
+    }
+    else {
+      this.city$
+      .pipe(
+        filter(city => !!city),
+        takeUntil(this.destroy$))
+      .subscribe(() => this.store.dispatch(new GetTopWorkshops(Constants.ITEMS_PER_PAGE)));          
+    }
+  }
 
   ngOnInit(): void {
-    this.store.dispatch([
-      new GetTopDirections(),
-      new GetTopWorkshops(Constants.ITEMS_PER_PAGE)
-    ]);
+    this.store.dispatch(new GetTopDirections());
 
-    this.actions$.pipe(ofAction(SetCity))
+    this.isParent$
       .pipe(
-        debounceTime(500),
-        distinctUntilChanged(),
         takeUntil(this.destroy$))
-      .subscribe(() => this.store.dispatch(new GetTopWorkshops(Constants.ITEMS_PER_PAGE)));
+      .subscribe(parent => this.parent = parent);
 
-    this.isParent$
+    this.role$
       .pipe(
+        filter(role => !!role),
         takeUntil(this.destroy$))
-      .subscribe(parent => this.parent = parent);
+      .subscribe(role => this.getTopWorkshops(role))  
   }
```

Comment:
I truly believe you don't need it anymore. It is not used in the ts. The "parent" is passed to the workshop card and as a boolean (that is not correct). So we do not really need the subscribtion to the parent. The parent input should be removed and the logic for the view should be replaces with the role-state. 
![image](https://user-images.githubusercontent.com/27493959/136608482-c0838cb6-2442-4fc8-8a55-ecefc24b3209.png)